### ISSUE TITLE
[Bug][a11y] The download dialogue and notification aren't read

### ISSUE THREAD

**Issue by softvision-miralobontiu** [mozilla-mobile/fenix#16050]
## Steps to reproduce
1. Turn Voice assistant/talkback on.
2. Go to https://file-examples.com/ and select to download various files, using TB gestures.

### Expected behavior
The download dialogues and the notifications are read/announced.

### Actual behavior
After selecting which file to download, the download dialogue which asks the user to confirm the download isn't read.
While the download is in progress, the notification with the download is not read/announced.
If the download fails, the failed download dialogue isn't read.

However, when the download is complete, **the complete download dialogue is read**.

### Device information
* Android device: Samsung Galaxy Tab S6 (Android 10)
* Fenix version: 10/20 Nightly, RC 82.1.0-RC.2, Beta 82.0.0-beta.6
 
<img src="https://user-images.githubusercontent.com/40694158/96559368-a15cc180-12c5-11eb-9f7d-224aef1d171e.png" width="300" height="550" />

**Comment by gouri-panda** [comment]
@Amejia481 I want to work on this issue, but I am confused about what to change? Can you tell me what to change? Thanks!

**Comment by Amejia481** [comment]
Thanks @gouri-panda, 
We have multiple dialogs that are not been read by the [TalkBack assistant](https://support.google.com/accessibility/android/answer/6283677), we have to investigate each case separately and find out what is the issue:

* Confirm download dialog, the [code](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/SimpleDownloadDialogFragment.kt) and [XML view](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/res/layout/mozac_downloads_prompt.xml). 

* Notification for an active download [code](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadNotification.kt).

* Third party download app chooser [code](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/ui/DownloadAppChooserDialog.kt).

Something that is interesting is that the completed download dialog is getting read  [see DynamicDownloadDialog.kt](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/downloads/DynamicDownloadDialog.kt)

If you need any additional help just let me know!

**Comment by gouri-panda** [comment]
@Amejia481 Apologies for the delay. I can't reproduce this bug. Can guide me on how to reproduce this bug? Thanks!

**Comment by Amejia481** [comment]
Following the steps on https://github.com/mozilla-mobile/fenix/issues/16050#issue-725329689 doesn't simulate the bug? If so which Android version are using?

**Comment by gouri-panda** [comment]
@Amejia481 Thanks for helping. I'm using Android 10, Manufactured by Realme. Is this happens only on Samsung? Samsung always changes android's API.

**Comment by Amejia481** [comment]
I tested on Android 11 and I wasn't able to replicated, @softvision-miralobontiu is this only happening on Samsung Galaxy Tab S6 (Android 10)?

**Comment by softvision-miralobontiu** [comment]
I was able to reproduce this issue on a Google Pixel (Android 10) device on the 2/4 Nightly build.

**Comment by gouri-panda** [comment]
@Amejia481 Any tips on how to reproduce this issue on pixel emulator. I can't on the text to speech feature in emulator.Thanks!

**Comment by Amejia481** [comment]
Sorry, for the delay try to use the an emulator with the same Android OS version and with google play. Pixel devices comes with stock Android, if it's reproducible in pixel phone is likely to be reproducible in emulator too.

**Comment by lobontiumira** [comment]
Reproducible on Beta 90.0.0-beta.3 with OnePlus 5T (Android 10).

**Comment by lobontiumira** [comment]
I was able to reproduce this issue on the 11/2 Nightly build with Sony Xperia Z5 Premium (Android 7.1.1).
